modules/azure/src/main/java/org/testcontainers/azure/AzureEventhubsEmulatorContainer.java

```diff
@@ -0,0 +1,106 @@
+package org.testcontainers.azure;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+/**
+ * Testcontainers implementation for Azure Eventhubs Emulator.
+ * <p>
+ * Supported image: {@code "mcr.microsoft.com/azure-messaging/eventhubs-emulator"}
+ * <p>
+ * Exposed ports:
+ * <ul>
+ *     <li>AMQP: 5672</li>
+ *     <li>Kafka: 9092</li>
+ * </ul>
+ */
+public class AzureEventhubsEmulatorContainer extends GenericContainer<AzureEventhubsEmulatorContainer> {
+
+    private static final int DEFAULT_AMQP_PORT = 5672;
+
+    private static final int DEFAULT_KAFKA_PORT = 9092;
+
+    private static final String CONNECTION_STRING_FORMAT =
+        "Endpoint=sb://%s:%d;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse(
+        "mcr.microsoft.com/azure-messaging/eventhubs-emulator"
+    );
+
+    private AzuriteContainer azuriteContainer;
+
+    private MountableFile config;
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public AzureEventhubsEmulatorContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        waitingFor(Wait.forLogMessage(".*Emulator Service is Successfully Up!.*", 1));
+        withExposedPorts(DEFAULT_AMQP_PORT, DEFAULT_KAFKA_PORT);
+    }
+
+    @Override
+    public void start() {
+        if (this.azuriteContainer == null) {
+            this.azuriteContainer =
+                new AzuriteContainer(AzuriteContainer.DEFAULT_IMAGE_NAME.withTag("3.33.0")).withNetwork(getNetwork());
+        }
+        this.azuriteContainer.start();
+
+        super.start();
+    }
```

Comment:
Do not override `start()`. You can declare Azurite on `dependsOn` in the constructor. However, my concern would be the Azurite fixed version 🤔

Reply:
Thanks for the remarks! I have a couple of questions:
1. I wanted to make sure the containers are on the same network, but in the constructor it is too early to copy. Should I skip setting the network or add the dependency maybe in `cofigure()` rather?
2. The Azurite version is an interesting puzzle because we need another parameter if we want to ensure that people can override it. Would you be fine if I added another `DockerImageName` parameter to let the user define which Azurite image they want to use? I guess that is the cleanest because they can use custom repos or compatible images etc.

Reply:
Just realized, that configure is too late because the dependencies are started before.

Reply:
I think, in the meantime, we can advocate to let the user declare the network and AzuriteContainer. This is one of the topic that depends on design.